`azurerm_app_service_connection`: add support for `scope`, `configuration_info` and `public_network_solution` (#26387)
terraform-providers/terraform-provider-azurerm: website/docs/r/app_service_connection.html.markdown

---

> * `scope` - (Optional) Connection scope in source service.

Reviewer: Is it possible to get the valid values here to assist users in completing this part of the config?

Author: I'll double check with the service team.

Author: Added some more info provided by the service team.

> * `scope` - (Optional) Connection scope in source service. It's namespace name if source is AKS cluster, container name if source is Container app.

Reviewer: I think this reads more clearly for the user grammatically. 
```suggestion
* `scope` - (Optional) Connection scope in the source service. This is the Namespace Name if the source is an AKS cluster, or the Container Name if source is a Container App.
```

Author: Sure, will fix.

* `configuration_info` - (Optional) The connection information consumed by applications, including secrets, connection strings. A `configuration_info` block as defined below.

---

> A `configuration_info` block supports the following:

Reviewer 2: info is redundent can we remove it?
```suggestion
A `configuration` block supports the following:
```

Author: Removed.

> * `configuration` - (Optional) The connection information consumed by applications, including secrets, connection strings. A `configuration` block as defined below.

Reviewer: grammar
```suggestion
* `configuration` - (Optional) The connection information consumed by applications, including secrets and connection strings. A `configuration` block as defined below.
```

Author: Will fix.

* `action` - (Optional) The action indicates whether to apply configurations on source application. Possible values are `enable` and `optOut`.

> * `configuration_store` - (Optional) A `configuration_store` block as defined below.

Reviewer: We seem to be missing the `configuration_store` block documentation?

Author: Will add.
